eclass/distutils-r1.eclass

```diff
@@ -1812,7 +1812,13 @@ distutils-r1_run_phase() {
 	local -x AR=${AR} CC=${CC} CPP=${CPP} CXX=${CXX}
 	tc-export AR CC CPP CXX
 
-	if [[ ${DISTUTILS_EXT} ]]; then
+	# Perform additional environment modifications only for python_compile
+	# phase.  This is the only phase where we expect to be calling the Python
+	# build system.  We want to localize the altered variables to avoid them
+	# leaking to other parts of multi-language ebuilds.  However, we want
+	# to avoid localizing them in other phases, particularly
+	# python_configure_all, where the ebuild may wish to alter them globally.
+	if [[ ${DISTUTILS_EXT} && ( ${1} == *compile* || ${1} == *test* ) ]]; then
```

Comment:
I think we should probably specifically check that this is distutils-r1_python_compile, and not a user defined `python_compile` or `python_compile_all`?

I guess I don't know whether someone would actually try to set up environment variables during a compile eclass-phase-hook instead of the configure hook.

Reply:
No. If you redefine `python_compile()`, you want things to work also. We're not expecting people to copy this whole logic into every ebuild defining a custom `python_compile`.